### PR TITLE
chore: enable deleting groups

### PR DESCRIPTION
**Pull request by magnew**
Closes: #8430 

### Description:

Enable deleting groups from the UI. 

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/UserSettings/UsersAndGroupsPanel/GroupsView.tsx [hunk 1]
```diff
@@ -1,23 +1,123 @@
...
+                </Group>
+            }
+        >
+            <Text pb="md">Are you sure you want to delete this group?</Text>
+            <Group spacing="xs" position="right">
+                <Button
+                    disabled={disableControls}
+                    onClick={onClose}
+                    variant="outline"
+                    color="dark"
```
Comment (joaoviana) [inline packages/frontend/src/components/UserSettings/UsersAndGroupsPanel/GroupsView.tsx:82]: Can we remove this? We don't do `color="dark"` for most `Cancel` buttons in Modals across the app. Unless I'm wrong!

**Review by joaoviana — APPROVED** [review]
Works great!

I would expect to see an empty state message when there's no groups - can be done later of course 👍 

image

And left a non-blocking comment

File: packages/frontend/src/components/UserSettings/UsersAndGroupsPanel/GroupsView.tsx (hunk 1 above)
Comment (magnew) [reply packages/frontend/src/components/UserSettings/UsersAndGroupsPanel/GroupsView.tsx:82]: Keeping it this way for now as discussed in slack

**Comment by magnew** [comment]
> Works great!
> 
> I would expect to see an empty state message when there's no groups - can be done later of course 👍
> 
> image
> And left a non-blocking comment

Coming soon!

**Pull request merged**